src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -307,28 +308,45 @@ class SlickDataAccess(databaseConfig: Config, val dataAccess: DataAccessComponen
     runTransaction(action)
   }
 
-  override def getWorkflowsByState(states: Traversable[WorkflowState]): Future[Traversable[WorkflowInfo]] = {
+  override def getWorkflow(workflowId: WorkflowId): Future[WorkflowDescriptor] = {
+    val action = for {
+      workflowExecutionResult <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
```

Comment:
@mcovarr on blanket or not, my thought was that we should start making raw heads and gets and such a little bit painful for the author to make them think twice about doing it. This is another one of those things which has hit a tipping point where both the use is increasing and people can point to "everyone else is doing it". We should be trying to reduce the number of these, not add to them.

However for here and now I think what @scottfrazer just proposed is fine.